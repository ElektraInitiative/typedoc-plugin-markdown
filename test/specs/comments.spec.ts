import * as Handlebars from 'handlebars';
import { ProjectReflection } from 'typedoc';

describe(`Comments:`, () => {
  let project: ProjectReflection;
  let partial: Handlebars.TemplateDelegate;
  beforeAll(async () => {
    project = await global.bootstrap(['comments.ts'], {
      includes: './test/stubs/inc',
      media: './test/stubs/media',
    });
    partial = global.getPartial('comment');
  });

  test(`should build @link references'`, () => {
    expect(
      global.compileTemplate(
        partial,
        project.getChildByName('commentWithDocLinks'),
      ),
    ).toMatchSnapshot();
  });

  test(`should convert symbols brackets to symbol links'`, () => {
    expect(
      global.compileTemplate(
        partial,
        project.getChildByName('commentsWithSymbolLinks'),
      ),
    ).toMatchSnapshot();
  });

  test(`should convert comments with fenced block'`, () => {
    expect(
      global.compileTemplate(
        partial,
        project.getChildByName('commentsWithFencedBlock'),
      ),
    ).toMatchSnapshot();
  });

  test(`should convert comments with tags'`, () => {
    expect(
      global.compileTemplate(
        partial,
        project.getChildByName('commentsWithTags'),
      ),
    ).toMatchSnapshot();
  });

  test(`should allow html in comments'`, () => {
    expect(
      global.compileTemplate(
        partial,
        project.getChildByName('commentsWithHTML'),
      ),
    ).toMatchSnapshot();
  });

  test(`should convert comments with includes'`, () => {
    expect(
      global.compileTemplate(
        partial,
        project.getChildByName('commentsWithIncludes'),
      ),
    ).toMatchSnapshot();
  });
});
