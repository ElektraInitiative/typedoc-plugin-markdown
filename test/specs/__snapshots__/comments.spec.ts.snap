// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments: should allow html in comments' 1`] = `
"<p>
You can write <strong>HTML</strong> tags directly in comments
</p>
<ul>
<li>List item</li>
</ul>
"
`;

exports[`Comments: should build @link references' 1`] = `
"See [\`CommentClass\`](CommentClass.md) and CommentClass's [comment](CommentClass.md#comment) property.
Also, check out [https://www.google.com|Google](https://www.google.com|Google) and
[GitHub](https://github.com).

Taken from [JsDoc](http://usejsdoc.org/tags-inline-link.html).
"
`;

exports[`Comments: should convert comments with fenced block' 1`] = `
"Some comments with fence blocks

\`\`\`typescript
someFunction()
\`\`\`

\`\`\`js
anotherFunction()
\`\`\`
"
`;

exports[`Comments: should convert comments with includes' 1`] = `
"This is an example of include

This is an example of handlebars include
"
`;

exports[`Comments: should convert comments with tags' 1`] = `
"**\`Description\`**

Tag description on new line

- Tag description on another line

**\`Deprecated\`**

Another tag description

Gets the application version.

**\`Example\`**

\`\`\`typescript
import { getVersion } from '@tauri-apps/api/app';
const appVersion = await getVersion();
\`\`\`
"
`;

exports[`Comments: should convert symbols brackets to symbol links' 1`] = `
"Additionally you can link to other classes, members or functions using double square brackets.

- Link to an external reflection: [CommentClass](CommentClass.md)
- Link to an internal reflection: [commentsInReturn](../modules.md#commentsinreturn)
- Link to an undefined reflection: VOID
"
`;
